packages/server/modules/gatekeeper/repositories/billing.ts

```diff
@@ -212,15 +213,55 @@ export const getWorkspaceSubscriptionBySubscriptionIdFactory =
     return subscription ?? null
   }
 
-export const getWorkspaceSubscriptionsPastBillingCycleEndFactory =
+const newPlans = Object.values(PaidWorkspacePlansNew)
+const oldPlans = Object.values(PaidWorkspacePlansOld)
+
+export const getWorkspaceSubscriptionsPastBillingCycleEndFactoryOldPlans =
   ({ db }: { db: Knex }): GetWorkspaceSubscriptions =>
   async () => {
     const cycleEnd = new Date()
     cycleEnd.setMinutes(cycleEnd.getMinutes() + 5)
     return await tables
       .workspaceSubscriptions(db)
-      .select()
+      .join(
+        WorkspacePlans.name,
+        WorkspacePlans.col.workspaceId,
+        'workspace_subscriptions.workspaceId'
+      )
+      .whereIn(WorkspacePlans.col.name, oldPlans)
       .where('currentBillingCycleEnd', '<', cycleEnd)
+      .select([
+        'workspace_subscriptions.workspaceId',
```

Comment:
Why not just `WorkspaceSubscriptions.cols` or even 'workspace_subscriptions.*'? I don't think the .select() is needed if you're selecting everything, it probably defaults to '*'

same goes for query below